pkg/build/controller/config_controller.go

```diff
@@ -55,9 +55,15 @@ func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) erro
 	}
 
 	glog.V(4).Infof("Running build for BuildConfig %s/%s", bc.Namespace, bc.Name)
+
+	buildTriggerCauses := []buildapi.BuildTriggerCause{}
 	// instantiate new build
 	lastVersion := 0
 	request := &buildapi.BuildRequest{
+		TriggeredBy: append(buildTriggerCauses,
+			buildapi.BuildTriggerCause{
+				Reason: "Triggered by config change",
```

Comment:
let me explain what i had in mind:
Triggered By is more of a header for the trigger causes. Whenever the serial builds PR is finished and we have more builds in queue(if i understand correctly) we have multiple causes, it will looks something like this:

```
[vagrant@origindevel origin]$ oc describe build ruby-sample-build-18
Name:           ruby-sample-build-18
Created:        3 hours ago
Labels:         app=ruby-sample-build
                buildconfig=ruby-sample-build
                name=ruby-sample-build
                openshift.io/build-config.name=ruby-sample-build
                template=application-template-dockerbuild
Annotations:    openshift.io/build.number=18
                openshift.io/build.pod-name=ruby-sample-build-18-build

Status:         Failed
Started:        Fri, 15 Apr 2016 18:05:17 UTC
Duration:       20s
Build Config:   ruby-sample-build
Build Pod:      ruby-sample-build-18-build

Strategy:               Docker
URL:                    git://github.com/openshift/ruby-hello-world.git
Commit:                 9bdc3a2 (Added license)
Author/Committer:       Anonymous User
From Image:             DockerImage centos/ruby-22-centos7@sha256:7452a984e755c78cca67df7763eff78a2e070b55c91da3cdb4704d128b7fad77
Output to:              ImageStreamTag origin-ruby-sample:latest
Post Commit Hook:       ["<image-entrypoint>", "bundle", "exec", "rake", "test"]
Push Secret:            builder-dockercfg-vudj6

Trigger causes:
Reason:         Build triggered by Generic webhook
Commiter:       Anonymous User, anonUser@example.com
Author:         Anonymous User, anonUser@example.com
Commit:         9bdc3a26ff933b32f3e558636b58aea86a69f051
Message:        Added license

Reason:                 Build triggered by GitHub webhook
Commit:                 9bdc3a2 (Added license)
Author/Committer:       Anonymous User

Reason:         Triggered by image change
Image Name:     ruby-22-centos7:latest
Kind:           ImageStreamTag

Reason:  Triggered by build configuration change

No events.
```
 @mfojtik I admit i might have misunderstood what your PR is about, been focusing more on this than whatever is around. :)